t/06elements.t

```diff
@@ -0,0 +1,605 @@
+# $Id$
+
+##
+# this test checks the DOM element and attribute interface of XML::LibXML
+
+use v6.c;
+use Test;
+
+#use base qw(ParentClass);
+#
+#sub new {
+#    my $class = shift;
+#    $class = ref $class if ref $class;
+#    my $self = bless {}, $class;
+#    $self;
+#}
+
+#1;
+
+# Should be 187.
+#use Test::More tests => 191;
+plan 187;
+
+
+use XML::LibXML;
+
+my $foo       = "foo";
+my $bar       = "bar";
+my $nsURI     = "http://foo";
+my $prefix    = "x";
+my $attname1  = "A";
+my $attvalue1 = "a";
+my $attname2  = "B";
+my $attvalue2 = "b";
+my $attname3  = "C";
+
+# TEST:$badnames=4;
+my @badnames= ("1A", "<><", "&", "-:");
+
+# 1. bound node
+{
+    my $doc = XML::LibXML::Document.new();
+    my $elem = $doc.createElement( $foo );
+    # TEST
+    ok $elem, 'created element';
+    # TEST
+    is $elem.tagName, $foo, 'element has correct tag name';
+
+    {
+        for @badnames -> $name {
+            try { 
+                my $caught = 0;
+                CATCH {
+                    ok $@, "setNodeName throws an exception for $name";
```

Comment:
I am thinking that if the `CATCH` was executed that we end up with two tests per iteration, otherwise one.
So I think that either the resume needs to go, or the test in the catch needs to go or the second test needs to happen conditionally.
I think I believe the second way might be the best.